File: 0386-lexicographical-numbers/0386-lexicographical-numbers.java
class Solution {
    public static List<Integer> res = new ArrayList<Integer>();
    public static void f(int i,int n){
        if(i<=n){
            if(i!=0)
              res.add(i);
        }else{
            return;
        }
        for(int k=0;k<=9;k++){
            if(k==0 && i==0) continue;
            f(10*i+k,n);
        }
    }
    public List<Integer> lexicalOrder(int n) {
        res.clear();
        f(0,n);
        return res;
    }
}

File: 0386-lexicographical-numbers/0386-lexicographical-numbers.cpp
class Solution {
public:
    void f(int i,int n,vector<int> &res){
        if(i>n) return;
        res.push_back(i);
        for(int j=0;j<=9;j++){
            int nn=i*10+j;
            if(nn>n) return;
            f(nn,n,res);
        }
    }
    vector<int> lexicalOrder(int n) {
        vector<int> res;
        for(int i=1;i<=9;i++){
          f(i,n,res);
        }
        return res;
    }
};